modules/CmpdReg/src/server/routes/CmpdRegRoutes.coffee

```diff
@@ -856,8 +857,35 @@ exports.updateParent = (req, resp) ->
 			resp.statusCode = 500
 			resp.end "Error trying to update parent: " + error;
 	)
-	
-	
+
+exports.swapParentStructures = (req, resp) ->
+	cmpdRegCall = config.all.client.service.cmpdReg.persistence.fullpath + '/parents/swapParentStructures'
+
+	username = req.session.passport?.user.username
+	if !username
+		resp.statusCode = 401
+		resp.end "No user name specified"
+		return
```

Comment:
I am not sure if I need to do this since the user is authenticated.

Reply:
I don't think this is necessary, because the user will already be authenticated.